M2/Macaulay2/m2/matrix.m2

```diff
@@ -3,6 +3,22 @@
 
 needs "modules.m2"
 
+notsamering := (X,Y) -> (
+     if X === Y then error("expected ",pluralsynonym X, " for the same ring")
+     else error("expected ",synonym X," and ",synonym Y," for the same ring"))
+samering = (M,N) -> if ring M === ring N then (M,N) else notsamering(class M,class N)
+nottosamering := (X,Y) -> (
+     if X === Y then error("expected ",pluralsynonym X, " for compatible rings")
+     else error("expected ",synonym X," and ",synonym Y," for compatible rings"))
+tosamering = (M,N) -> (
```

Comment:
I don't mind keeping the old names if you don't want to change names in a bunch of places, but I don't think we should actively rename something camel-cased to all lower case. Also, maybe move these functions to where `isPromotable` is defined instead.

Reply:
will do. not the most urgent thing right now.
once again, this is a *draft*.